prow/plugins/project/project_test.go

```diff
@@ -0,0 +1,394 @@
+			repo:            "kubernetes",
+			org:             "kubernetes",
+			commenter:       "sig-lead",
+			previousProject: "0.1.0",
+			previousColumn:  "To do",
+			expectedProject: "0.1.0",
+			expectedColumn:  "To do",
+			expectedError:   fmt.Sprintf(invalidProject, []string{"kubernetes/*", "kubernetes/kubernetes"}),
+		},
+	}
+
+	fakeClient := &fakegithub.FakeClient{
+		RepoProjects:      repoProjects,
+		ProjectColumnsMap: projectColumnsMap,
+		ColumnIDMap:       columnIDMap,
+		IssueComments:     make(map[int][]github.IssueComment),
+	}
+	for _, tc := range testcases {
+		fakeClient.Project = tc.previousProject
+		fakeClient.Column = tc.previousColumn
+
+		e := &github.GenericCommentEvent{
+			Action: github.GenericCommentActionCreated,
+			Body:   tc.body,
+			Number: 1,
+			Repo:   github.Repo{Owner: github.User{Login: tc.org}, Name: tc.repo},
+			User:   github.User{Login: tc.commenter},
+		}
+
+		if err := handle(fakeClient, logrus.WithField("plugin", pluginName), e, projectConfig); err != nil {
+			if tc.expectedError != err.Error() {
+				t.Errorf("Expected error %s but got %s", tc.expectedError, err.Error())
+				continue
+			}
+			t.Errorf("(%s): Unexpected error from handle: %v.", tc.name, err)
+			continue
+		}
+
+		if fakeClient.Project != tc.expectedProject ||
+			fakeClient.Column != tc.expectedColumn {
+			t.Errorf("\nFor test %s,\nExpected Project=%s Column=%s, but got Project=%s Column=%s", tc.name, tc.expectedProject, tc.expectedColumn, fakeClient.Project, fakeClient.Column)
+		}
+	}
+}
+
+func TestParseCommand(t *testing.T) {
+	var testcases = []struct {
+		command         string
+		proposedProject string
+		proposedColumn  string
+		shouldClear     bool
+	}{
+		{
+			command:         "/project 0.0.0 To do",
+			proposedProject: "0.0.0",
+			proposedColumn:  "To do",
+			shouldClear:     false,
+		},
+		{
+			command:         "/project 0.0.0 Backlog",
+			proposedProject: "0.0.0",
+			proposedColumn:  "Backlog",
+			shouldClear:     false,
+		},
+		{
+			command:         "/project clear 0.0.0",
+			proposedProject: "0.0.0",
+			proposedColumn:  "",
+			shouldClear:     true,
+		},
+		{
+			command:         "/project clear",
+			proposedProject: "",
+			proposedColumn:  "",
+			shouldClear:     false,
+		},
+		{
+			command:         "/project 0.0.0",
+			proposedProject: "0.0.0",
+			proposedColumn:  "",
+			shouldClear:     false,
+		},
+	}
+
+	for _, test := range testcases {
+		proposedProject, proposedColumn, shouldClear, _ := parseCommand(test.command)
+		if proposedProject != test.proposedProject ||
+			proposedColumn != test.proposedColumn ||
+			shouldClear != test.shouldClear {
+			t.Errorf("\nFor command %s, expected\n  proposedProject = %s\n  proposedColumn = %s\n  shouldClear = %t\nbut got\n  proposedProject = %s\n  proposedColumn = %s\n  shouldClear = %t\n", test.command, test.proposedProject, test.proposedColumn, test.shouldClear, proposedProject, proposedColumn, shouldClear)
```

Comment:
Would be nicer for the person who breaks this test to check them separately and do tailored messaged to each

Reply:
Or wrap them in a struct and use ObjectReflectDiff or something